### ISSUE TITLE
[Feature]: Parental controls

### ISSUE THREAD

**Issue by imgalaxyhunter** [simplex-chat/simplex-chat#4482]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

all

### Feature

Some people are going to find this unpopular but I think it's necessary as people are going to use unsafe third party solutions to achieve the same thing, which often requires downgrading to SMS as there's no way for an app access messages in third party messaging apps app on iOS at all, so parental control service might login to iCloud account to receive SMS/iMessages instead, on Android it would require an invasive accessibility service recording everything done on the phone.

Now that's out of the way, here's the two features I'm proposing.

**1)** Ability to have messages automatically forwarded to another contact in a way where they can essentially see what you're seeing, this could be an extension of a more adequate linked devices system, but I think parents shouldn't have the ability to send messages as their child, that seems too far to me.

**2)** Parental control PIN, this would be a prerequisite to what I described above because the child could just remove the linked device/parental controls otherwise. This could be used to also disallow/require approval before adding contacts, approval could be done remotely or require the PIN to be entered on the device. This could be security theatre though as on Android you can just clear the app storage and create a new profile without parental controls, I'm not sure if Google or Apple offers a way to stop this with their parental controls, I wouldn't know as I don't use any of their services.

**Comment by imgalaxyhunter** [comment]
And it should go without saying, it must be obvious that parental controls are enabled, this should never be allowed to be done in secret and there should always be a 🔴recording icon visible at the top of the app or something.

**Comment by imgalaxyhunter** [comment]
There could also be the ability to add time limits, etc which can be handled by OS parental controls, but I think it would be good to have this built into the app also for people who aren't using Google or Apple services on their devices.

**Comment by imgalaxyhunter** [comment]
> just don't let the kids install it and don't embarrass yourself

What else are they going to use to communicate?

**Comment by imgalaxyhunter** [comment]
I am not for or against it, but the reality is people ARE going to use invasive parent monitoring apps so I would prefer if it was handled securely by the app itself.

**Comment by SHUVAMETA** [comment]
Don't let kids use this platform for communication. Simple!
This application meant to be free (free means freedom), adding parental control will be a stab to this free(dom)

**Comment by imgalaxyhunter** [comment]
> Don't let kids use this platform for communication. Simple! This application meant to be free (free means freedom), adding parental control will be a stab to this free(dom)

right, I get that, but the alternative is much worse.. if parents are going to do this anyway then why not limit the data access to the parent and child instead of offloading to some sketchy third party service with poor security practices?

**Comment by kimg45** [comment]
Forwarding messages to another party is antithetical to privacy.

**Comment by imgalaxyhunter** [comment]
> Forwarding messages to another party is antithetical to privacy.
> 
> Your argument that parents are going to do it anyway is sort of like saying “they’re going to get a virus anyway, might as well pre install the virus” If people intentionally ruin their own software it’s their fault.

This is about limiting data access. Parent spyware services are widely used, we need privacy focused versions of everything, it's what the market demands.

**Comment by m6n50u7** [comment]
> just don't let the kids install it and don't embarrass yourself

yeah, let them use other apps that spy on them for the big tech groomers. this is better than parental control.

**Comment by m6n50u7** [comment]
parental control would be good to not let those pro censorship use the "think about the children" card every time to get the support of the people.
